app/services/random_author.rb

```diff
@@ -8,12 +8,12 @@ def call(genre = nil)
                        <<~SQL.squish,
                          exists (
                            select 1 from
-                             creations c
-                             join works w on (c.work_id = w.id)
+                             involved_authorities ia
```

Comment:
again, don't we lose the distinction here, if we don't filter by role?  This could now return someone who only *translated* a work in a particular genre.

Reply:
Yes. you're right. Fixed. 

If we need to get only authors here we need to add filter by role. By the way, old code, also could return illustrators. I believe it was mitigated by the fact that only person with TOC were selected, but I've added explicit filter here.